driver-core/src/test/java/com/datastax/driver/core/FrameLengthTest.java

```diff
@@ -0,0 +1,128 @@
+            create.addColumn("col" + i, DataType.blob());
+        }
+        execute(create.getQueryString());
+
+        // build prepared statement.
+        Insert insert = insertInto(tableName).value("k", bindMarker()).value("c", bindMarker());
+        for (int i = 0; i < colCount; i++) {
+            insert = insert.value("col" + i, bindMarker());
+        }
+
+        PreparedStatement prepared = session().prepare(insert);
+
+        // Insert rows.
+        logger.info("Inserting data for {} partitions.", partitionCount);
+        for (int i = 0; i < partitionCount; i++) {
+            logger.info("Inserting {} rows in partition {}", rowsPerPartitionCount, i);
+            for (int r = 0; r < rowsPerPartitionCount; r++) {
+                BoundStatement stmt = prepared.bind();
+                stmt.setInt("k", i);
+                stmt.setInt("c", r);
+                for (int c = 0; c < colCount; c++) {
+                    byte[] b = new byte[bytesPerCol];
+                    random.nextBytes(b);
+                    ByteBuffer in = ByteBuffer.wrap(b);
+                    stmt.setBytes("col" + c, in);
+                }
+                session().execute(stmt);
+            }
+        }
+        logger.info("Done loading {}", tableName);
+    }
+
+    /**
+     * Validates that if a frame is received that exceeds NATIVE_TRANSPORT_MAX_FRAME_SIZE_IN_MB that
+     * the driver is able to recover and reconnect to each host and make subsequent queries.  It
+     * configures NATIVE_TRANSPORT_MAX_FRAME_SIZE_IN_MB to 1 MB to make the error easier to reproduce.
+     *
+     * @jira_ticket JAVA-1292
+     * @jira_ticket JAVA-1293
+     * @test_category connection
+     */
+    @Test(groups = "isolated")
+    public void should_throw_exception_when_frame_exceeds_configured_max() {
+        // Capture the connection logger to check for TooLongFrameException.
+        org.apache.log4j.Logger connectionLogger = org.apache.log4j.Logger.getLogger(Connection.class);
+        Level originalLevel = connectionLogger.getLevel();
+        connectionLogger.setLevel(Level.DEBUG);
+        MemoryAppender logs = new MemoryAppender();
+        connectionLogger.addAppender(logs);
+        try {
+            session().execute(select().from(tableName).where(eq("k", 0)));
+            fail("Exception expected");
+        } catch (NoHostAvailableException e) {
+            // Both hosts should have been tried.
+            assertThat(e.getErrors()).hasSize(2);
+        } finally {
+            // Check that TooLongFrameException is encountered as it isn't raised to the NoHostAvailableException.
+            assertThat(logs.get()).contains("io.netty.handler.codec.TooLongFrameException");
+            connectionLogger.setLevel(originalLevel);
+            connectionLogger.removeAppender(logs);
+        }
+        // Connection will be lost for both hosts, but they should be able to reconnect.
+        waitForUp(ipOfNode(1), cluster(), 10);
```

Comment:
I think I can work around by doing the following:

1. After the first message, we  assume the version of all frames will be the same on the same connection.  This way we don't have to parse version every time.
2. Check the opcode after creating the frame.  The Frame body is not parsed until later anyways, so this should be ok, the disadvantage of this is that we may fail later on in the process. I.E. if the payload is malformed and the full frame length is never received, the request would just time out instead.  That might not be too bad if the data is bad anyways.